R/BASS_studyArea_analysis.R

```diff
@@ -1,157 +1,96 @@
-#' @param idcol Identification column for units
-#' @param hexid Column for hexagon identification
-#' @param removedhexes Vector of names of hexagons to remove
-#' @param return_points
+#' @param probs Data frame. Output of `calculate_inclusion_probs()` or
+#'   `full_BASS_run()`.
+#' @param nARUs Numeric vector. Number of samples or named vector of number of
+#'   samples per stratum.
+#' @param os Numeric vector. Over sample size (proportional) or named vector of
+#'   number of samples per stratum.
+#' @param removed_hexes Character Vector. Ids of hexagons to remove.
+#'
+#' @inheritParams common_docs
 #'
 #' @return
 #' @export
 #'
-run_grts_on_BASS <- function(n_grts_tests, study_area_results, nARUs, os,
-                             idcol, hexid, removedhexes = c("None"), return_points = F, Stratum = None, ...) {
-  if (as_label(enquo(hexid)) == "<empty>") stop("run_grts_on_BASS now requires you to specify hexagon column under hexid. Please correct and try again.")
+run_grts_on_BASS <- function(probs, num_runs, nARUs, os,
+                             hex_id = NULL,
+                             stratum_id = NULL,
+                             remove_hexes = NULL) {
 
-  if(packageVersion("spsurvey")<5){
-    if (is.list(study_area_results) & !is.data.frame(study_area_results)) {
-      if (!is_null(study_area_results$inclusionPr)) {
-        attframe <- study_area_results$inclusionPr
-      }
-      if (is_null(study_area_results$inclusionPr)) {
-        fbr_t <- study_area_results %>% transpose()
-        attframe <- fbr_t %>%
-          .[["inclusion_probs"]] %>%
-          do.call("rbind", .) %>%
-          as_tibble() %>%
-          dplyr::select(-geometry) %>%
-          filter(!is.na(X))
-      }
-    } else {
-      if ("geometry" %in% names(study_area_results)) {
-        attframe <- as_tibble(study_area_results) %>%
-          dplyr::select(-geometry) %>%
-          filter(!is.na(X))
-      } else {
-        attframe <- as_tibble(study_area_results)
-      }
-    }
-    attframe <- filter(attframe, !{{ hexid }} %in% removedhexes)
-  if(as_label(enquo(Stratum)) == "None"){
-  Stratdesgn <- rep(list(PanelOne = list # a list named 'None" that contains:
-  (
-    panel = c(PanelOne = rep(nARUs)),
-    over = rep(nARUs * os), # panelOne indicates the number of samples you want
-    seltype = "Continuous"
-  )), length(unique(attframe[[idcol]])))
-  names(Stratdesgn) <- unique(attframe[[idcol]])
-  } else{
-    l_s <- unique(attframe[[idcol]])
-      Stratdesgn <-
-        purrr::map(l_s, ~list # a list named 'None" that contains:
-                             (
-                               panel = c(PanelOne =  nARUs$N[nARUs[[idcol]] == .x],
-                               over = max(round(nARUs$N[nARUs[[idcol]] == .x] * os,0),1)), # panelOne indicates the number of samples you want
-                               seltype = "Continuous" )
-                             )
-      names(Stratdesgn) <- l_s
-  }
+  mindis <-  NULL
```

Comment:
I'm guessing I'll probably want to pull these two default values into variable names rather than hiding them in the code.